paimon-core/src/main/java/org/apache/paimon/metrics/commit/CommitMetrics.java

```diff
@@ -0,0 +1,180 @@
+                    return latestCommit == null ? 0L : latestCommit.getNumBucketsWritten();
+                });
+        genericMetricGroup.histogram(COMMIT_DURATION, durationHistogram);
+        genericMetricGroup.gauge(
+                LAST_TABLE_FILES_ADDED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getTableFilesAdded();
+                });
+        genericMetricGroup.gauge(
+                LAST_TABLE_FILES_DELETED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getTableFilesDeleted();
+                });
+        genericMetricGroup.gauge(
+                LAST_TABLE_FILES_APPENDED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getTableFilesAppended();
+                });
+        genericMetricGroup.gauge(
+                LAST_TABLE_FILES_COMMIT_COMPACTED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getTableFilesCompacted();
+                });
+        genericMetricGroup.gauge(
+                LAST_CHANGELOG_FILES_APPENDED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getChangelogFilesAppended();
+                });
+        genericMetricGroup.gauge(
+                LAST_CHANGELOG_FILES_COMMIT_COMPACTED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getChangelogFilesCompacted();
+                });
+        genericMetricGroup.gauge(
+                LAST_DELTA_RECORDS_APPENDED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getDeltaRecordsAppended();
+                });
+        genericMetricGroup.gauge(
+                LAST_CHANGELOG_RECORDS_APPENDED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getChangelogRecordsAppended();
+                });
+        genericMetricGroup.gauge(
+                LAST_DELTA_RECORDS_COMMIT_COMPACTED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getDeltaRecordsCompacted();
+                });
+        genericMetricGroup.gauge(
+                LAST_CHANGELOG_RECORDS_COMMIT_COMPACTED,
+                () -> {
+                    return latestCommit == null ? 0L : latestCommit.getChangelogRecordsCompacted();
+                });
+        totalTableFilesCounter = genericMetricGroup.counter(TOTAL_TABLE_FILES);
+        totalChangelogFilesCounter = genericMetricGroup.counter(TOTAL_CHANGELOG_FILES);
+    }
+
+    public void reportCommit(CommitStats commitStats) {
+        latestCommit = commitStats;
+        totalTableFilesCounter.inc(
+                commitStats.getTableFilesAdded() - commitStats.getTableFilesDeleted());
+        totalChangelogFilesCounter.inc(
+                commitStats.getChangelogFilesAppended() + commitStats.getChangelogFilesCompacted());
```

Comment:
In PIP-3 you say that `totalTableFiles` is the "umber of total data files currently maintained on storage". However this implementation only counts the number of table files created during this job.